### ISSUE TITLE
MetricsRequestHandler expects certain endpoint name length

### ISSUE THREAD

**Issue by rsvoboda** [smallrye/smallrye-metrics#73]
https://github.com/smallrye/smallrye-metrics/blob/master/implementation/src/main/java/io/smallrye/metrics/MetricsRequestHandler.java#L78 (`String scopePath = requestPath.substring(8);`) expects certain length of the name of the endpoint which exposes metrics data.

So when I attach metrics data to `/me` endpoint I get following results

```bash
curl  -H "Accept: application/json" 'http://localhost:8080/me/trics'
Bad scope requested: sScope s not found

curl  -H "Accept: application/json" 'http://localhost:8080/me/tricvendor'
{
  "thread.count" : 18,
  "memory.heap.usage" : 154778168,
  "memory.nonHeap.usage" : 55497072
}

curl  -H "Accept: application/json" 'http://localhost:8080/me'
   Internal Server Error
server throws:
java.lang.StringIndexOutOfBoundsException: String index out of range: -5
	at java.lang.String.substring(String.java:1931)
	at io.smallrye.metrics.MetricsRequestHandler.handleRequest(MetricsRequestHandler.java:71)
```

**Comment by jmartisk** [comment]
How and why would you exactly do that? :)
- AFAIK in WildFly the context `/metrics` is hardcoded https://github.com/wildfly/wildfly/blob/15.0.0.Final/microprofile/metrics-smallrye/src/main/java/org/wildfly/extension/microprofile/metrics/MetricsContextService.java#L53
- in Thorntail as well https://github.com/thorntail/thorntail/blob/2.3.0.Final/fractions/microprofile/microprofile-metrics/src/main/java/org/wildfly/swarm/microprofile/metrics/runtime/MetricsHttpHandler.java#L56
- and MP Metrics spec requires this too https://github.com/eclipse/microprofile-metrics/blob/1.1.1/spec/src/main/asciidoc/metrics_spec.adoc#23-exposing-metrics-via-rest-api

The problem is that there is no mechanism in place in Thorntail or WildFly to tell SR-Metrics what the context root is, so SR-Metrics has to guess. I agree that the guessing could have been done better, I guess no one expected that the context can be different than the usual. To make this reliable, I think the only way would be to introduce a mechanism in Thorntail and WildFly to convey the context to SR-Metrics. Is this really a use case we need to support?

**Comment by rsvoboda** [comment]
I think it's nice to have, I just don't like hard-coded `requestPath.substring(8)`.
Some guessing could be introduced.

**Comment by kenfinnigan** [comment]
There's actually an open JIRA for Thorntail to make it customizable.

I don't think that's unreasonable, so we should look to handle it here somewhow

**Comment by jmartisk** [comment]
I have a potential fix here: https://github.com/smallrye/smallrye-metrics/pull/74 It adds a new variant of `MetricsRequestHandler.handleRequest` which accepts a custom context root as a string, the old variant will assume `/metrics`

**Comment by jmartisk** [comment]
Merged for 2.0
